### PR TITLE
Jit64: Make DoubleToSingle a common asm routine

### PR DESCRIPTION
**Pull request by booto**

### CODE REVIEW

**Comment by MerryMage** [comment]
This can be implemented without needing `XMM2`:

```
void CommonAsmRoutines::GenConvertDoubleToSingle()
{
  // Input in XMM0, output to XMM0
  // Clobbers RSCRATCH/RSCRATCH2/XMM1

  const void* start = GetCodePtr();

  // Grab Exponent
  MOVQ_xmm(R(RSCRATCH), XMM0);
  MOV(64, R(RSCRATCH2), R(RSCRATCH));
  SHL(64, R(RSCRATCH), Imm8(52));
  AND(32, R(RSCRATCH), Imm32(double_exponent >> 52));

  // Check if the double is in the range of valid single subnormal
  SUB(16, R(RSCRATCH), Imm16(874));
  CMP(16, R(RSCRATCH), Imm16(896 - 874));
  FixupBranch NoDenormalize = J_CC(CC_A);

  // Denormalise

  // shift = (905 - Exponent) plus the 21 bit double to single shift
  MOV(16, R(RSCRATCH), Imm16(905 + 21));
  MOVQ_xmm(XMM1, R(RSCRATCH));
  PSUBQ(XMM1, R(XMM0));

  // XMM0 = fraction | 0x0010000000000000
  MOVAPD(XMM0, R(XMM2));
  PAND(XMM0, MConst(double_fraction));
  POR(XMM0, MConst(double_explicit_top_bit));

  // fraction >> shift
  PSRLQ(XMM0, R(XMM1));

  // OR the sign bit in.
  SHL(64, R(RSCRATCH2), 32);
  AND(32, R(RSCRATCH2), Imm32(0x80000000));
  MOVQ_xmm(XMM1, R(RSCRATCH));

  POR(XMM0, R(XMM1));
  RET();

  // Don't Denormalize
  SetJumpTarget(NoDenormalize);

  // We want bits 0, 1
  MOVAPD(XMM1, R(XMM0));
  PAND(XMM1, MConst(double_top_two_bits));
  PSRLQ(XMM1, 32);

  // And 5 through to 34
  PAND(XMM0, MConst(double_bottom_bits));
  PSRLQ(XMM0, 29);

  // OR them togther
  POR(XMM0, R(XMM1));
  RET();

  JitRegister::Register(start, GetCodePtr(), "JIT_cdts");
}
```

File: Source/Core/Core/PowerPC/Jit64/Jit_LoadStoreFloating.cpp [hunk 1]
```diff
@@ -115,7 +115,8 @@ void Jit64::stfXXX(UGeckoInstruction inst)
     {
       RCX64Reg Rs = fpr.Bind(s, RCMode::Read);
       RegCache::Realize(Rs);
-      ConvertDoubleToSingle(XMM0, Rs);
+      MOVSD(XMM0, R(Rs));
```
Comment (MerryMage) [inline Source/Core/Core/PowerPC/Jit64/Jit_LoadStoreFloating.cpp]: MOVAPD(XMM0, Rs);

**Comment by booto** [comment]
Thanks for the implementation without XMM2. Applied that now, with some minor fix ups:

SHL(64, R(RSCRATCH), Imm8(52)); -> SHR(64, R(RSCRATCH), Imm8(52));

AND(32, R(RSCRATCH), Imm32(double_exponent >> 52)); -> AND(16, R(RSCRATCH), Imm16(0x7ff));

SHR(64, R(RSCRATCH2), 32); -> SHR(64, R(RSCRATCH2), Imm8(32));

**Comment by booto** [comment]
See #8120 instead

**Pull request closed without merging**